File: Python/백준/Silver/10845. 큐/큐.py
import sys
class queue:
    def __init__(self):
        self.container = ['']
        self.head = 0
    def __len__(self):
        return len(self.container) - self.head - 1
    def pop(self):
        self.head += 1
        return self.container[self.head]
    def push(self, n):
        self.container.append(n)
    def value(self, n):
        if len(self) == 0:
            return -1
        if n >= 0:
            return self.container[self.head + n + 1]
        else:
            return self.container[n]
    
file = queue()
N = int(sys.stdin.readline())
for _ in range(N):
    inp = tuple(map(str, (sys.stdin.readline().rstrip() + " 1").split()))
    if inp[0] == 'push':
        file.push(int(inp[1]))
    elif inp[0] == 'front':
        print(file.value(0))
    elif inp[0] == 'back':
        print(file.value(-1))
    elif inp[0] == 'empty':
        print(int(not bool(len(file))))
    elif inp[0] == 'size':
        print(len(file))
    elif inp[0] == 'pop':
        if len(file):
            print(file.pop())
        else:
            print(-1)
        
        
    

File: Java/백준/Silver/10845. 큐/큐.java
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.BufferedReader;
import java.io.OutputStreamWriter;
import java.io.BufferedWriter;
import java.util.*;
import java.lang.Math;

public class Main {
    static InputStreamReader reader = new InputStreamReader(System.in);
    static BufferedReader br = new BufferedReader(reader);

    static OutputStreamWriter writer = new OutputStreamWriter(System.out);
    static BufferedWriter bw = new BufferedWriter(writer);
    public static void main(String[] args) throws IOException {
        int opCnt = Integer.parseInt(br.readLine());
        Queue q = new Queue(10001);

        for (int i = 0; i < opCnt; i++) {
            String inp[] = br.readLine().split(" ");
            if (inp[0].equals("push")) {
                q.push(Integer.parseInt(inp[1]));
            }
            else if (inp[0].equals("pop")) {
                bw.write(Integer.toString(q.pop()) + "\n");
            }
            else if (inp[0].equals("size")) {
                bw.write(Integer.toString(q.size()) + "\n");
            }
            else if (inp[0].equals("empty")) {
                bw.write(Integer.toString(q.isEmpty()) + "\n");
            }
            else if (inp[0].equals("front")) {
                bw.write(Integer.toString(q.front()) + "\n");
            }
            else if (inp[0].equals("back")) {
                bw.write(Integer.toString(q.back()) + "\n");
            }

        }
        bw.flush();
        bw.close();
    }
}
class Queue {
    private int container[];
    private int maxSize = 0;

    private int startIdx = 0;
    private int endIdx = 0;
    private int currentItemsCnt = 0;
    
    Queue(int size) {
        this.maxSize = size;
        this.container = new int[size];
    }

    void push(int X) {
        this.container[this.endIdx] = X;
        this.endIdx = (this.endIdx + 1) % this.maxSize;
        this.currentItemsCnt += 1;
    }   

    int isEmpty() {
        return (currentItemsCnt == 0)?1:0;
    }

    int size() {
        return currentItemsCnt;
    }

    int pop() {
        if (this.isEmpty() == 1) {
            return -1;
        } else {
            int popped = this.container[this.startIdx];
            this.startIdx = (this.startIdx + 1) % this.maxSize;
            this.currentItemsCnt -= 1;
            return popped;
        }
    }

    int front() {
        if (this.isEmpty() == 1) return -1;
        else return this.container[this.startIdx];
    }

    int back() {
        if (this.isEmpty() == 1) return -1;
        else {
            int newEnd = (this.endIdx - 1) % this.maxSize;
            return this.container[newEnd];
        }
    }
}